amaranth/sim/pysim.py

```diff
@@ -229,17 +271,78 @@ def commit(self):
         return awoken_any
 
 
+_PyMemoryChange = namedtuple("_PyMemoryChange", ["state", "addr"])
+
+
+class _PyMemoryState(BaseMemoryState):
+    __slots__ = ("memory", "data", "pending_writes", "waiters", "pending")
```

Comment:
`write_queue`? we have two completely different things with `pending` in the name now